provider/pkg/provider/ids.go

```diff
@@ -73,5 +86,11 @@ func getDefaultName(
 	if err != nil {
 		return resource.PropertyValue{}, err
 	}
+
+	// Apply naming trivia to the generated name.
+	if namingTriviaApplies {
+		random = namingTrivia.ApplyTrivia(sdkName, random)
+	}
```

Comment:
Sort of a code style question - do we prefer adding explicit branches in the outer layer like this, or would having the `ApplyTrivia` method on `NamingTrivia` be a no-op in the case where we've decided it doesn't apply make more sense?  I tend to prefer not to add hidden complexity when possible, but am open to arguments otherwise.

Reply:
I think this is fine given this existing structure of the code. There might be some scope to restructure this a little more in the future. Possibly calculating the random length after getting the trivia. Similarly, the validation of the length could be done after putting the whole id together for simplicity - so we could include the example generated id in the error message so the user can see how it's constructed. This isn't really needed to get this feature out though - just making notes while I'm parsing.